sklearn/discriminant_analysis.py

```diff
@@ -113,6 +113,14 @@ def _class_means(X, y):
     """
     xp, is_array_api_compliant = get_namespace(X)
     classes, y = xp.unique_inverse(y)
+    # Force lazy array api backends to call compute
+    if hasattr(classes, "persist"):
+        classes.compute_chunk_sizes()
+        # We get a ValueError: no field named inverse later on if we don't
+        # compute right now
+        # Probably a dask bug
+        # (the error is also kinda flaky)
+        y = y.compute()
```

Comment:
To shed more light on what this is, I think this bug happens when scikit-learn calls unique_inverse.

I think something is going wrong in dask somewhere where the results of intermediate operations are getting corrupted.

When the error occurs in computation is somewhat flaky, but it happens more often than not without the compute here.

Reply:
Did you investigate the cause of the corruption? It might be worth reporting a minimal reproducer upstream.